homeassistant/components/stt/__init__.py

```diff
@@ -156,3 +160,31 @@ def _metadata_from_header(request: web.Request) -> SpeechMetadata:
         )
     except TypeError as err:
         raise ValueError(f"Wrong format of X-Speech-Content: {err}") from err
+
+
+@websocket_api.websocket_command(
+    {
+        "type": "stt/engine/list",
+        vol.Optional("language"): str,
+    }
+)
+@callback
+def websocket_list_engines(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """List speech to text engines and, optionally, if they support a given language."""
+    legacy_providers: dict[str, Provider] = hass.data[DOMAIN]
+
+    providers = {
+        "providers": [
+            {
+                # entity_id in preparation for migration of stt engines to entities
+                "entity_id": provider,
```

Comment:
This isn't currently am entity_id. Should we instead list both types, legacy and entity_id separately and add a type item?

Reply:
I was considering that, but I'm not sure it matters. We could just call it `engine_id` and let stt resolve to a legacy or modern engine?

Reply:
I don't think we have had rules for how the provider name string must be set. So I don't think we can resolve safely to either entity_id or provider name just based on the characters of the engine id.

Reply:
I guess we can just try both.